cmd/root.go

```diff
@@ -31,8 +31,8 @@ func init() {
 	rootCmd.PersistentFlags().StringVarP(namespace, "namespace", "n", "", "namespace for environment")
 	rootCmd.PersistentFlags().StringP("file", "f", "", "Specify an alternate compose file")
 	envID := new(string)
-	rootCmd.PersistentFlags().StringVarP(envID, "env-id", "e", "", "used to isolate environments deployed to a shared namespace, "+
-		"by (1) using this value as a suffix of pod and service names and (2) using this value to isolate selectors(required)")
+	rootCmd.PersistentFlags().StringVarP(envID, "env-id", "e", "", "(required) used to isolate environments deployed to a shared namespace, "+
```

Comment:
Should set this flag to be required

`rootCmd.PersistentFlags().StringVarP(envID, "env-id", "e", "", "(required) used to isolate environments deployed to a shared namespace, "+
		"by (1) using this value as a suffix of pod and service names and (2) using this value to isolate selectors(required)")			"by (1) using this value as a suffix of pod and service names and (2) using this value to isolate selectors")
rootCmd. MarkPersistentFlagRequired("env-id")`

If it's set but with no value then check ENV vars?

Reply:
I just changed the message, anyway does setting it to required make cobra fail if there is no flag but there is an environment variable?

Reply:
fails if the flag is not set, I think if it's required the flag should be set, then we validate either the input or the ENV var.

Reply:
Otherwise, it's confusing saying it's required but we don't set it required until after we check its non-empty or no ENV var??

Reply:
I just looked at the code, and looks like the flag is made required only if the environment variable is not set.